x-pack/plugins/apm/server/routes/transactions/get_latency_charts/index.ts

```diff
@@ -72,7 +76,10 @@ function searchLatency({
   });
 
   const transactionDurationField =
-    getDurationFieldForTransactions(documentType);
+    documentType === ApmDocumentType.ServiceTransactionMetric &&
+    latencyAggregationType !== LatencyAggregationType.avg
+      ? TRANSACTION_DURATION_HISTOGRAM
+      : getDurationFieldForTransactions(documentType);
```

Comment:
Not needed for this PR but:
This logic is not trivial to grok, and having to duplicate this and several places will likely lead to bugs eventually when someone else come across and modify in 6 months. Would be great if we can think of some better abstractions to handle this

Reply:
Agreed!